### PR TITLE
deps: deduplicate dependencies

### PR DESCRIPTION
**Pull request by onur-ozkan**
## Description

Helps to reduce number of dependencies by removing duplicated ones.

### CODE REVIEW

File: transports/uds/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ tokio = { workspace = true, default-features = false, features = ["net"], option
 tracing = { workspace = true }
 
 [dev-dependencies]
-tempfile = "3.10"
+tempfile = "3.6"
```
Comment (onur-ozkan) [inline transports/uds/Cargo.toml]: Are we okay with this downgrade? I think it should be fine since it's a dev-dependency.

**Review by jxs — COMMENTED** [review]
Hi, and thanks for your interest! As we use dependabot to submit PR's with updates for the dependencies I don't think this is worth it.
If we downgrade `tempfile` dependabot will submit a PR trying to update it afaik, trying to keep track of downgraded packages and keeping ignoring upgrades for certain so that we have less package versions on `Cargo.lock` is therefore not worth it imho.

**Comment by onur-ozkan** [comment]
> Hi, and thanks for your interest! As we use dependabot to submit PR's with updates for the dependencies I don't think this is worth it. If we downgrade `tempfile` dependabot will submit a PR trying to update it afaik, trying to keep track of downgraded packages and keeping ignoring upgrades for certain so that we have less package versions on `Cargo.lock` is therefore not worth it imho.

Only the 2nd commit downgrades a dependency as pointed out at https://github.com/libp2p/rust-libp2p/pull/5318#discussion_r1568777634. I will revert that commit since the repo has dependabot integration.

**Review by jxs — DISMISSED** [review]
Thanks!

**Comment by drHuangMHT** [comment]
Ping @jxs for stalled PR

**Review by jxs — APPROVED** [review]

**Pull request merged**